packages/koenig-lexical/src/components/ui/cards/CodeBlockCard.jsx

```diff
@@ -1,29 +1,75 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import useAutoExpandTextArea from '../../../utils/autoExpandTextArea';
+import CodeMirror from '@uiw/react-codemirror';
+import {EditorView} from '@codemirror/view';
 
-export function CodeBlockCard({code, language, updateCode, updateLanguage}) {
-    const textareaRef = React.useRef(null);
+export function CodeEditor({code, language, updateCode, updateLanguage}) {
+    const onChange = React.useCallback((value) => {
+        updateCode(value);
+    }, [updateCode]);
 
-    useAutoExpandTextArea({el: textareaRef, value: code});
+    const editorCSS = EditorView.theme({
+        '.cm-content, .cm-gutter': {
+            minHeight: '170px'
+        },
+        '.cm-scroller': {
+            overflow: 'auto'
+        }
+    });
```

Comment:
@sanne-san heads up, the styling situation for the code editor (using 3rd party tool called codemirror) is a bit funky — it uses 'themes', which are basically CSS-in-JS. More info [here](https://codemirror.net/examples/styling/#themes).